### PR TITLE
BL-784 feature/parent_family/add_child_profile

### PR DESCRIPTION
**Pull request by anastasia-lapteva**
## Description

Jira Ticket # **BL-784** Feature: ParentFamilyHome (ParentFamily) should properly display a parent's children. We want to start with the basics of a parent adding a profile for their child, similar to how we see profiles in the popular streaming apps like Netflix and Disney+.

Co-Author: **Vasilii Garanin**

## Fixes

- ParentFamilyHome now displays a parent's children
- Updated getChildren function in parent actions for catch error payload to err.message
- Created initialState in parentReducer
- Removed hardcoded parentDummyData from parentReducer, and instead initializing with new initialState
- Updated useEffect in ParentFamilyHome, currently hardcoded with an existing seeded profile_id, and left a comment for future programmers to update this once profile_id is properly passed along in currentUser (userActions/ userReducer)
- Also included idToken in InstructorNewsFeed NewsContainer (which is a previous ticket we worked on, included this update to  maintain page functionality following Okta version update)

## Loom Video

We were not able to fit full explanation into 5 minutes, so will link 2 videos below:

- [1] https://www.loom.com/share/9fd75fe4807e4fb693bdfff3f8d31c29
- [2] https://www.loom.com/share/248cecab16044a52a5eb4ac0f9b036ff

## Type of change

- [X] New feature (non-breaking change which adds functionality)

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have removed unnecessary comments/console logs from my code
- [X] I have made corresponding changes to the documentation if necessary (optional)
- [X] My changes generate no new warnings
- [X] I have checked my code and corrected any misspellings
- [X] No duplicate code left within changed files
- [X] Size of pull request kept to a minimum
- [X] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by Jwatk13 — COMMENTED** [review]
Great loom video explanation of your work! Overall good job going through and getting the parent children profiles setup! Your work is easy to read, understandable, and concise!

File: src/redux/actions/parentActions.js [hunk 1]
```diff
@@ -20,14 +20,20 @@ export const CLEAR_CART = 'CLEAR_CART';
 export const GET_NEWSFEEDS_PARENT = 'GET_NEWSFEEDS_PARENT';
 
 export const getChildren = (idToken, profile_id) => async dispatch => {
-  dispatch({ type: GET_CHILDREN_ACTION });
+  console.log('getChildren profile_id:', profile_id);
```
[the remark this answers is not in the document]
Comment (anastasia-lapteva) [reply src/redux/actions/parentActions.js:23]: Thank you, we missed this one! Nice catch!

File: src/components/pages/ParentFamily/ParentFamilyHome.js [hunk 2]
```diff
@@ -33,16 +32,16 @@ const ParentFamilyHome = props => {
...
 
   const dispatch = useDispatch();
   const { user, children } = props;
 
   useEffect(() => {
-    oktaAuth.token.getUserInfo().then(dataProfile => {
-      dispatch(getChildren(authState.idToken.idToken, dataProfile.sub));
-    });
-  }, [dispatch, oktaAuth.token, authState.idToken.idToken]);
+    // In the following line, "4" is hardcoded instead of profile_id because currently profile_id is not being passed along in props.user . profile_id 4 exists in our seeds, so it has been hardcoded to display the existing seeded child. currentUser is initializing as empty object {} and is not being updated. When this is fixed, change the hardcoded "4" to "profile_id" and it should work.
```
[the remark this answers is not in the document]
Comment (anastasia-lapteva) [reply src/components/pages/ParentFamily/ParentFamilyHome.js:42]: Thank you that's a great point! I will add in TODO keyword

File: src/redux/actions/parentActions.js [hunk 3]
```diff
@@ -20,14 +20,20 @@ export const CLEAR_CART = 'CLEAR_CART';
...
+      dispatch({
+        type: GET_CHILDREN_SUCCESS,
+        payload: res.data,
+      });
     })
     .catch(err => {
-      dispatch({ type: ERROR_ACTION, payload: err });
+      dispatch({
+        type: ERROR_ACTION,
+        payload: err.message,
```
[the remark this answers is not in the document]
Comment (vasilii-garanin) [reply src/redux/actions/parentActions.js:34]: Thank you for all feedback, Joseph

**Review by Jwatk13 — APPROVED** [review]

**Comment by vasilii-garanin** [comment]
> Wow you did a great job explaining what yall did in yalls code! The news feed page looks a lot better and good idea leaving some notes in there for whoever works on it next!

Thank you for feedback, Caitlin. If you can please approve changes we would appreciate. If you click on "Files Changed" > green "Review Changes" button in upper right corner > select "Approve" > select green "Submit Review" :+1: Thanks again for your help

**Review by CaitlinTalbert — APPROVED** [review]

**Review by vasilii-garanin — APPROVED** [review]

**Comment by AlexiusThomas** [comment]
Great job explaining your process and breaking down the code. Even in  error, thank you for the additional corrective material.

**Review by AlexiusThomas — COMMENTED** [review]
Great job explaining your process and breaking down the code. Even in error, thank you for the additional corrective material.

**Pull request merged**